### ISSUE TITLE
Register to template provides correct anat_to_template warping field but incorrect inverse

### ISSUE THREAD

**Issue by benjamindeleener** [neuropoly/spinalcordtoolbox#274]
Data are located here: /Volumes/folder_shared/sct_issues/20150316_allan

To reproduce the issue:
```
sct_register_to_template -i t2r.nii.gz -l t2r-labels.nii.gz -m t2r_seg.nii.gz -s fast -r 0
```

**Comment by benjamindeleener** [comment]
The problem comes from sct_register_multimodal. It produces only the deformation field from straigth to template but not the inverse.

**Comment by benjamindeleener** [comment]
the warping field from template to straight anat seems to have only extremely high values (see screenshot below).

![screen shot 2015-03-16 at 10 18 24](https://cloud.githubusercontent.com/assets/7757117/6667974/f9c76c46-cbc5-11e4-8351-ceb2c76724be.png)
![screen shot 2015-03-16 at 10 18 28](https://cloud.githubusercontent.com/assets/7757117/6667975/fb47db14-cbc5-11e4-83bd-f25c203aecef.png)

**Comment by benjamindeleener** [comment]
The problem does not happen when the initial label is slightly different from above (see file /Volumes/folder_shared/sct_issues/20150316_allan/t2r-labels_2.nii.gz)

```
sct_register_to_template -i t2r.nii.gz -l t2r-labels2.nii.gz -m t2r_seg.nii.gz -s fast -r 0
```

Results of template registration here:
![screen shot 2015-03-16 at 11 22 24](https://cloud.githubusercontent.com/assets/7757117/6669347/c3fadc16-cbce-11e4-9167-611a819f69a3.png)

THE ISSUE IS NOT FIXED YET!

**Comment by benjamindeleener** [comment]
The issue seemed to be caused by SliceReg when the segmentation did not covered the whole image. In some case, the generated warping field is nearly nul everywhere but some small values (~ e^-308) cause instabilities when inverting warping fields.

Solution: when registering to the template, make sure that the segmentation covers the whole image by cropping the segmentation image with sct_crop_image with the option -bzmax and -dim (0,1 or 2, is the dimensio of Z). sct_crop_image integrates -bzmax option since commit 715491a.

**Comment by benjamindeleener** [comment]
Cropping is not a solution as we do not want to provide an output image that is not in the same space than the input image.
Another solution would be to crop the image for template registration but compute the warping field in the input image space. It probably does not work...

**Comment by benjamindeleener** [comment]
will be fixed through sct_register_multimodal (issue #265)
